cmd/nginx-ingress/main.go

```diff
@@ -445,6 +451,16 @@ func getSocketClient(sockPath string) *http.Client {
 	}
 }
 
+// validateLockName validates the name of a resource
+// It returns an error if the resource name is invalid
+func validateLockName(lock string) error {
+	lst := validation.IsDNS1123Subdomain(lock)
+	if len(lst) > 0 {
+		return fmt.Errorf("invalid resource name: %v", lock)
```

Comment:
Here we lose all context of the error returned from: `validation.IsDNS1123Subdomain(lock)`

Maybe something like:
```suggestion
		return fmt.Errorf("invalid resource name %v: %v", lock, allErrs)
```

Can work? As allErrs is a slice of strings.